tests/test_nitpicking.py

```diff
@@ -427,6 +427,110 @@ def some_method(self, test):
             if not is_py3 and path.endswith(".py"):
                 content = f"# -*- coding: utf-8 -*-\n{content}"
             assert Path(path).read_text() == content
+    # Make sure it doesn't fail for incorrect module version
+    with local.cwd(tmp_path / "odoo" / "custom" / "src" / "private"):
+        manifest = "__manifest__" if is_py3 else "__openerp__"
+        build_file_tree(
+            {
+                f"test_module_2/{manifest}.py": f"""\
+                    {"{"}
+                        "name": "test module 2",
+                        "license": "AGPL-3",
+                        "version": "{supported_odoo_version - 1}.1.0.0",
+                        "installable": False,
+                        "auto_install": False,
+                    {"}"}
+                """,
+                "test_module_2/__init__.py": """\
+                    from . import models
+                """,
+                "test_module_2/models/__init__.py": """\
+                    from . import res_partner
+                """,
+                "test_module_2/models/res_partner.py": '''\
+                    import io
+                    import sys
+                    from logging import getLogger
+                    from os.path import join
+
+                    from requests import get
+
+                    import odoo
+                    from odoo import models
+
+                    _logger = getLogger(__name__)
+
+
+                    class ResPartner(models.Model):
+                        _name = "res.partner"
+
+                        def some_method(self, test):
+                            """some weird
+                            docstring"""
+                            _logger.info(models, join, get, io, sys, odoo)
+                ''',
+            }
+        )
+        git("add", "-A")
+        # First commit will add new module to the exclude list in pre-commit
+        git("commit", "-m", "added test_module_2", retcode=1)
+        git("commit", "-am", "added test_module_2")
```

Comment:
I think you have to check that there are no changes inside the module that cannot be installed.